ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -848,43 +848,54 @@ translateLedgerStateShelleyToAllegraWrapper ::
+              -- to be deleted, and their balance has to be moved to the
+              -- reserves. For this matter, the Ledger keeps track of these
+              -- small set of entries since the Byron to Shelley transition and
+              -- provides them to us through 'shelleyToAllegraAVVMsToDelete'.
               --
-              -- Once the ledger keeps track of these values, this operation
-              -- will no longer be needed.
-              . stowLedgerTables
-              . flip withLedgerTables (ShelleyLedgerTables $ ApplyValuesMK $ HD.UtxoValues $ SL.unUTxO deadCardanoAVVM)
+              -- In the long run, the ledger will already use ledger states
+              -- parametrized by the map kind and therefore will already provide
+              -- the differences in this translation.
+              let avvms           = SL.unUTxO $ fromSMaybe
+                                      (error "Shelley to Allegra AVVMs are not tracked in the Ledger")
+                                      (shelleyToAllegraAVVMsToDelete $ shelleyLedgerState ls)
+
+                  -- While techically we can diff the LedgerTables, it becomes
+                  -- complex doing so, as we cannot perform operations with
+                  -- 'LedgerTables l1 mk' and 'LedgerTables l2 mk'. Because of
+                  -- this, for now we choose to generate the differences out of
+                  -- thin air and when the time comes in which ticking produces
+                  -- differences, we will have to revisit this.
+                  avvmsAsDeletions = ShelleyLedgerTables
+                                   . ApplyDiffMK
+                                   . HD.UtxoDiff
+                                   . Map.map (  (`HD.UtxoEntryDiff` HD.UedsDel)
+                                              . unTxOutWrapper
+                                              . SL.translateEra' ()
+                                              . TxOutWrapper
+                                             )
+                                   $ avvms
+
+                  -- This 'stowLedgerTables' + 'withLedgerTables' injects the
+                  -- values provided by the Ledger so that the translation
+                  -- operation finds those entries in the UTxO and destroys
+                  -- them, modifying the reserves accordingly.
+                  stowedState = stowLedgerTables
+                              . withLedgerTables ls
+                              . ShelleyLedgerTables
+                              . ApplyValuesMK
+                              . HD.UtxoValues
+                              $ avvms
+
+                  resultingState = unFlip . unComp
+                                 . SL.translateEra' ()
+                                 . Comp   . Flip
+                                 $ stowedState
+
+              in resultingState `withLedgerTables` avvmsAsDeletions
```

Comment:
There may be a simple answer to this that I could find with some care, but I'm just going to ask instead: Should this instead be combining `avvmsAsDeletions` with whatever was in `projectLedgerTables resultingState`?

Reply:
Technically yes, although for now it is only UTxO and we know the `projectLedgerTables resultingState` will be empty. I kinda noted that in the comment above, specifying that this is a special case.